base_layer/core/src/transactions/transaction_protocol/sender.rs

```diff
@@ -342,13 +340,18 @@ impl SenderTransactionProtocol {
     /// Performs sanitary checks on the collected transaction pieces prior to building the final Transaction instance
     fn validate(&self) -> Result<(), TPE> {
         if let SenderState::Finalizing(info) = &self.state {
-            let total_amount = info.calculate_total_amount();
+            let total_amount = calculate_total_amount(info.amount_to_self, info.amounts.clone());
             let fee = info.metadata.fee;
             // The fee should be less than the amount. This isn't a protocol requirement, but it's what you want 99.999%
-            // of the time, and our users will thank us if we reject a tx where they put the amount in the fee field by
-            // mistake!
+            // of the time, however, always preventing this will also prevent spending dust in some edge cases.
             if fee > total_amount {
-                return Err(TPE::ValidationError("Fee is greater than amount".into()));
+                warn!(
```

Comment:
This was a point of contention in my mind if it is possible to construct a transaction without calling ` pub fn build<D: Digest>(mut self, factories: &CryptoFactories)` (as in some tests), or alternatively, to somehow alter the data after the initial send to the receiver, so I opted to be conservative and keep the sanity check in both places similar to the other sanity checks (`MAX_TRANSACTION_INPUTS`, `MINIMUM_TRANSACTION_FEE`).

I am quite happy to drop the sanity check in `fn validate(&self)` if you feel strongly about it.

Reply:
I think because it's a fat-finger check and not a consensus critical check, we should just do it once, and before things hit the wire as Mike said (I thought that both calls were before transmission initially, so I didn't bring it up in my previous reviews).

Reply:
Keeping it in the builder only then

> Not for this PR, but I think some of the other checks in `validation` could be done before anything hits the wire as well, minimum fee for instance.

Sanity checks for these values were already done - `MAX_TRANSACTION_INPUTS` and `MINIMUM_TRANSACTION_FEE` - added `MAX_TRANSACTION_OUTPUTS`.